pkg/testing/ess/deployment.go

```diff
@@ -0,0 +1,403 @@
+			} `json:"kibana"`
+		} `json:"resources"`
+	}
+
+	if err := json.NewDecoder(getResp.Body).Decode(&getRespBody); err != nil {
+		return nil, fmt.Errorf("error parsing deployment retrieval API response: %w", err)
+	}
+
+	r.ElasticsearchEndpoint = getRespBody.Resources.Elasticsearch[0].Info.Metadata.ServiceUrl
+	r.KibanaEndpoint = getRespBody.Resources.Kibana[0].Info.Metadata.ServiceUrl
+
+	return &r, nil
+}
+
+// ShutdownDeployment attempts to shut down the ESS deployment with the specified ID.
+func (c *Client) ShutdownDeployment(ctx context.Context, deploymentID string) error {
+	u, err := url.JoinPath("deployments", deploymentID, "_shutdown")
+	if err != nil {
+		return fmt.Errorf("unable to create deployment shutdown API URL: %w", err)
+	}
+
+	res, err := c.doPost(ctx, u, "", nil)
+	if err != nil {
+		return fmt.Errorf("error calling deployment shutdown API: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		return fmt.Errorf("got unexpected response code [%d] from deployment shutdown API", res.StatusCode)
+	}
+
+	return nil
+}
+
+// DeploymentStatus returns the overall status of the deployment as well as statuses of every component.
+func (c *Client) DeploymentStatus(ctx context.Context, deploymentID string) (*DeploymentStatusResponse, error) {
+	getResp, err := c.getDeployment(ctx, deploymentID)
+	if err != nil {
+		return nil, fmt.Errorf("error calling deployment retrieval API: %w", err)
+	}
+	defer getResp.Body.Close()
+
+	var getRespBody struct {
+		Resources struct {
+			Elasticsearch []struct {
+				Info struct {
+					Status DeploymentStatus `json:"status"`
+				} `json:"info"`
+			} `json:"elasticsearch"`
+			Kibana []struct {
+				Info struct {
+					Status DeploymentStatus `json:"status"`
+				} `json:"info"`
+			} `json:"kibana"`
+			IntegrationsServer []struct {
+				Info struct {
+					Status DeploymentStatus `json:"status"`
+				} `json:"info"`
+			} `json:"integrations_server"`
+		} `json:"resources"`
+	}
+
+	if err := json.NewDecoder(getResp.Body).Decode(&getRespBody); err != nil {
+		return nil, fmt.Errorf("error parsing deployment retrieval API response: %w", err)
+	}
+
+	s := DeploymentStatusResponse{
+		Elasticsearch:      getRespBody.Resources.Elasticsearch[0].Info.Status,
+		Kibana:             getRespBody.Resources.Kibana[0].Info.Status,
+		IntegrationsServer: getRespBody.Resources.IntegrationsServer[0].Info.Status,
+	}
+	s.Overall = overallStatus(s.Elasticsearch, s.Kibana, s.IntegrationsServer)
+
+	return &s, nil
+}
+
+// DeploymentIsReady returns true when the deployment is ready, checking its status
+// every `tick` until `waitFor` duration.
+func (c *Client) DeploymentIsReady(ctx context.Context, deploymentID string, waitFor, tick time.Duration) (bool, error) {
+	timer := time.NewTimer(waitFor)
+	defer timer.Stop()
+
+	ticker := time.NewTicker(tick)
+	defer ticker.Stop()
+
+	statusCh := make(chan DeploymentStatus, 1)
+	errCh := make(chan error)
+
+	for tick := ticker.C; ; {
```

Comment:
I cribbed the implementation from [`assert.Eventually`](https://github.com/stretchr/testify/blob/c5fc9d6b6b21ea89be8480c0dc35e2977ab988f6/assert/assertions.go#L1801) since the whole `waitFor` + `tick` pattern was exactly the same.  I'll remove it.